packages/predicate/src/utils/getContractRoot.ts

```diff
@@ -14,19 +14,12 @@ export const getContractRoot = (bytecode: BytesLike, chainId: number): string =>
     chunks.push(chunk);
   }
 
-  const totalBytes = chunks.reduce((sum, chunk) => chunk.byteLength + sum, 0);
-  const lastChunk = chunks[chunks.length - 1];
-  const isDivisibleBy16 = totalBytes % 16 === 0;
-  const remainingBytes = chunkSize - lastChunk.length;
-  if (!isDivisibleBy16) {
-    const nearestMultiple = Math.ceil(remainingBytes / 8) * 8;
-    const paddedChunkLength = lastChunk.length + nearestMultiple;
-    const paddedChunk = new Uint8Array(paddedChunkLength).fill(0);
-    paddedChunk.set(lastChunk, 0);
-    chunks[chunks.length - 1] = paddedChunk;
-  }
-
   const chainIdBytes = uint64ToBytesBE(chainId);
+  const lastChunk = chunks[chunks.length - 1];
+  const remainingBytes = bytes.length % chunkSize;
+  const paddedChunkLength = remainingBytes + ((8 - (remainingBytes % 8)) % 8);
+  const newChunk = lastChunk.slice(0, paddedChunkLength);
+  chunks[chunks.length - 1] = newChunk;
```

Comment:
It looks like we have this logic repeated here and on the contract can we reuse it?

Reply:
Agreed, can we include that in follow ups to unblock anyone requiring 0.19.0? I'm planning on working on the follows immediately, there's a couple other ones in the description of this PR.